lib/reek/smell_detectors/syntax.rb

```diff
@@ -17,21 +17,16 @@ def self.smells_from_source(source)
 
       # :reek:FeatureEnvy
       def smells_from_source(source)
+        context = Struct.new(:exp, :full_name).new(
```

Comment:
This does mean creating two new classes every time this method is called. Maybe assign those two structs to constants?